### PR TITLE
Implement tagExists Maven command (Fix issue 1063)

### LINKED ISSUE #1063: tagExists command not available with Maven Plugin
**Description**
It's not possible to call tagExists method rfom maven plugin

**Expected behavior**
It shoud be possible to run this kind of command : mvn liquibase:tagexists that returns the existance of a tag

### PR DESCRIPTION
**Pull request by rozenshteyn**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Implement tagExists Maven command. Fixes #1063.

Modeled after the existing LiqubaseTag mojo class.
Users will be able to run the command:  `mvn liquibase:tagExists -Dliquibase.tag=myTag` to check for existence of the specified tag in the configured database.

## Things to be aware of

- None at this time

## Things to worry about

- I am not sure if anything else (other than implementing the mojo class) needs to be done in order for the functionality to show up in the Liquibase Maven plugin.

## Additional Context

None at this time.

### CODE REVIEW

**Comment by MalloD12** [comment]
Approved!

# Review and testing results:

Overall this PR looks ok to me. Few small changes has been made and are detailed below:

- Added goal annotation to **TagExists** Mojo class.
- Updated **testTag** unit tests to validate tag is set and properly set to the mojo under test.
- Update existent **testTag** integration test to also perform a **tagExists** validation. I decided to update the existing test and not added a new one, because the test would be pretty much the same and I thought it is worth using the same setup and not duplicate code for testing almost the same scenario.  

### Things to be aware of:
- Nothing

### Things to worry about:
- Nothing. Only applicable to Maven plugin so no further impact.


Additional comment: 
Although **autocandidate** tag has been added. I think might be worth performing a round of exploratory testing around maven plugin. We have unit/integration tests to validate commands are successfully executed, but at this point nothing maven specific.

**Review by MalloD12 — APPROVED** [review]

File: liquibase-maven-plugin/src/test/java/org/liquibase/maven/plugins/LiquibaseTagExistsMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Some basic tests that validate the setting of properties for the
+ * LiquibaseTagExistsMojo.
+ */
+public class LiquibaseTagExistsMojoTest extends AbstractLiquibaseMojoTest {
+
+	// reusing dropAll/plugin_config.xml since the LiguibaseTagExists mojo does
```
Comment (XDelphiGrl) [inline liquibase-maven-plugin/src/test/java/org/liquibase/maven/plugins/LiquibaseTagExistsMojoTest.java:15]: I appreciate the comment explaining why this particular changelog is used in the test. Thank you!

**Comment by XDelphiGrl** [comment]
@MalloD12, hi!
> Additional comment: Although **autocandidate** tag has been added. I think might be worth performing a round of exploratory testing around maven plugin. We have unit/integration tests to validate commands are successfully executed, but at this point nothing maven specific.

We do have functional automated tests that exercise Maven goals. We should add a new one for this feature. I will create an internal Jira ticket to make sure we do that. I do not believe this particular PR requires any additional manual testing.

**Comment by MalloD12** [comment]
> We do have functional automated tests that exercise Maven goals. We should add a new one for this feature. I will create an internal Jira ticket to make sure we do that. I do not believe this particular PR requires any additional manual testing.

Oh nice, I might not see them yet or might forget about those. Where are they? I

**Review by XDelphiGrl — APPROVED** [review]
This PR extends the Liquibase Maven plugin to support the `tagExists` goal. 
* Functional tests passing with a manual rerun here: [https://github.com/liquibase/liquibase-pro-tests/actions/runs/3473764788](https://github.com/liquibase/liquibase-pro-tests/actions/runs/3473764788)
  * An internal Liquibase functional test ticket was created to add a `tagExists` test. (DAT-12404)
* No additional testing necessary.

APPROVED

**Review by filipelautert — APPROVED** [review]
All fine, new maven command.

**Review by suryaaki2 — APPROVED** [review]

**Pull request merged**